float64 width
float64 force
---
bool success